packages/trader/src/App/Components/Elements/ContractDrawer/contract-drawer.jsx

```diff
@@ -40,7 +55,14 @@ class ContractDrawer extends Component {
         this.setState({ is_shade_on: shade });
     };
 
-    redirectBackToReports = () => this.props.history.push(routes.reports);
+    redirectBackToReports = () => {
+        // history.goBack() will go to the wrong location if user goes to contract by pasting it in the url.
+        if (this.props.history.location.state) {
```

Comment:
Just a query, are we planning to redirect back to anywhere apart from `reports` route?
Because currently if user goes to `contract-replay` from `positions-drawer` it shouldn't have a back button.

Reply:
I don't know, I did not change the functionality, I might have missed a merge conflict, if the removed line is the correct behavior.